lookup_plugins/barman_server.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: barman_server
+    author: Julien Tachoires
+    short_description: Lookup Barman server, based on its private_ip
+    description:
+      - "Lookup Barman server, based on its private_ip"
+    options:
+      _terms:
+        description: Private IP of the Barman server.
+        required: False
+      default:
+        description: barman_server_private_ip of the current node.
+"""
+
+EXAMPLES = """
+- name: Show Barman server informations
+  debug: msg="{{ lookup('barman_server') }}"
+"""
+
+RETURN = """
+_value:
+  description:
+    - List of Barman server nodes
+  type: list
+  elements:
+    - dict: node_type, hostname, ansible_host (public IP address), private_ip
+"""
+
+from ansible.errors import AnsibleError
+from ansible.inventory.manager import InventoryManager
+from ansible.plugins.lookup import LookupBase
+
+
+class LookupModule(LookupBase):
+    def run(self, terms, variables=None, **kwargs):
+
+        myvars = getattr(self._templar, '_available_variables', {})
+        # Inventory hostname
+        ihn = variables['inventory_hostname']
```

Comment:
Can we change the variable name to something meaningful?

Reply:
Done.